src/mesh/boundary_info.C

```diff
@@ -1806,6 +1806,78 @@ BoundaryInfo::build_node_list_from_side_list()
      node_bcid_action_functor, datum_type_ex);
 }
 
+void BoundaryInfo::sync_push_remove_boundary_side_id(
+  std::unordered_map<processor_id_type, std::vector<std::pair<dof_id_type, unsigned int>>>
+  & elems_to_push)
+{
+  auto elem_action_functor =
+    [this]
+    (processor_id_type,
+     const std::vector<std::pair<dof_id_type, unsigned int>> & received_elem)
+    {
+      for (const auto & pr : received_elem)
+        this->remove_side(_mesh.elem_ptr(pr.first), pr.second);
+    };
+
+  Parallel::push_parallel_vector_data
+    (this->comm(), elems_to_push, elem_action_functor);
```

Comment:
Where you are using `sync_push_remove_boundary_side_id` @dewenyushu ? I thought boundary sides could be removed locally (without communication). And then `sync_pull_boundary_side_id` should update ghosted elements correctly. 

I might be missing something here.

Reply:
The push is needed in my application because I need to remove the neighbor element side from the boundary. The neighbor element is often a ghost element so I need to communicate this change by push. 

Roy and I are curious if you may need similar functionalities in your application (i.e., remove/add sides for the ghost elements)? So we could discuss whether this function should be kept in libmesh.